### ISSUE TITLE
Agent WeaveExecutor fails with dead weaveexec containers on CoreOS beta, alpha

### ISSUE THREAD

**Issue by SpComb** [kontena/kontena#1526]
This started to happen on my CoreOS Beta 1235.2.0 vagrant node recently. Seen something similar before with Ubuntu xenial, and switching to the overlayfs2 volume driver fixed it.

```
kontena-agent | E, [2016-12-09T11:12:50.978477 #1] ERROR -- : Actor crashed!
kontena-agent | Docker::Error::ServerError: Driver overlay failed to remove root filesystem 9835a3da45f02f0d0f000ffbb0221612a36d4f233c950e4a49aba52b1797995e: remove /var/lib/docker/overlay/b492d218b9b7ab546b11b24dc7f46aef107c264e425e15b5ef7a0ff364bc20f8/merged: device or resource busy
kontena-agent | 
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/docker-api-1.32.1/lib/docker/connection.rb:50:in `rescue in request'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/docker-api-1.32.1/lib/docker/connection.rb:38:in `request'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/docker-api-1.32.1/lib/docker/connection.rb:65:in `block (2 levels) in <class:Connection>'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/docker-api-1.32.1/lib/docker/container.rb:234:in `remove'
kontena-agent | 	/app/lib/kontena/network_adapters/weave_executor.rb:79:in `execute'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/call/sync.rb:16:in `dispatch'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
kontena-agent | 	/usr/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
```

**Comment by SpComb** [comment]
Something similar also happens sporadically with the `weaveexec attach --rewrite-hosts` itself, when weve itself runs docker containers internally:

```
kontena-agent | D, [2016-12-09T11:19:08.230834 #1] DEBUG -- NetworkAdapters::WeaveExecutor: weaveexec ok: ["--local", "attach", "10.81.128.112/16", "--rewrite-hosts", "5ca6f11adc73f9223ba9ea6921f76ecb0b857a069b3d1e2251230b61231247ab"]
kontena-agent | 10.81.128.112
kontena-agent | 
kontena-agent | D, [2016-12-09T11:19:09.617610 #1] DEBUG -- NetworkAdapters::WeaveExecutor: weaveexec ok: ["--local", "attach", "10.81.128.17/16", "--rewrite-hosts", "a56d0c3d2f2cfd0180435a638cf64c37a575323488714c5f1d38d6615db86631"]
kontena-agent | 10.81.128.17
kontena-agent | 
kontena-agent | E, [2016-12-09T11:19:09.840137 #1] ERROR -- Kontena::NetworkAdapters::WeaveExecutor: weaveexec exit 1: ["--local", "attach", "10.81.128.93/16", "--rewrite-hosts", "8778203da2517e55a5afc252f110e701a9ab027964b83d4900f28a9c03e2d520"]
kontena-agent | time="2016-12-09T11:19:09Z" level=fatal msg="Error response from daemon: Driver overlay failed to remove root filesystem d713608499a1c91e5145c01ad4db24709a889367949a954a43435dff283d5051: remove /var/lib/docker/overlay/3099ba8be0ea57137e7da99cca6eb0df07d6eb9404ca3d6ed4fb99bd9544908a/merged: device or resource busy" 
kontena-agent | 
kontena-agent | E, [2016-12-09T11:19:11.041305 #1] ERROR -- Kontena::NetworkAdapters::WeaveExecutor: weaveexec exit 1: ["--local", "attach", "10.81.128.71/16", "--rewrite-hosts", "418f5d6c2e5270201b189a61a5081b60defed9265e58751f88eccd3a1534c80b"]
kontena-agent | time="2016-12-09T11:19:10Z" level=fatal msg="Error response from daemon: Driver overlay failed to remove root filesystem 56fe5db3e3b21e49df4cf688a60bc33c53ad3262920dde73ffef62b7660ff85d: remove /var/lib/docker/overlay/6d75b338837fac0fc3961d11c1f313f874b199795825ebbe598247628dc0119d/merged: device or resource busy" 
kontena-agent | 
kontena-agent | E, [2016-12-09T11:19:11.260270 #1] ERROR -- Kontena::NetworkAdapters::WeaveExecutor: weaveexec exit 1: ["--local", "attach", "10.81.128.72/16", "--rewrite-hosts", "7383374b2e003123f8d78f4a11cf4653bce183641b80793b0ac6fd7acc5fe034"]
kontena-agent | time="2016-12-09T11:19:11Z" level=fatal msg="Error response from daemon: Driver overlay failed to remove root filesystem eb0e2352b1720103bc10fb5380a1c8166aa89e26784f201c2694b78339e9d44d: remove /var/lib/docker/overlay/dcf447337b29cf681b8bf81b60cc6f139e3999967324b612951873d27d00509d/merged: device or resource busy" 
kontena-agent | 
kontena-agent | D, [2016-12-09T11:19:12.279116 #1] DEBUG -- NetworkAdapters::WeaveExecutor: weaveexec ok: ["--local", "attach", "10.81.128.55/16", "--rewrite-hosts", "2e124a2590efee15c26c37f8babc006367be4a5d25402c42020e775d36391246"]
kontena-agent | 10.81.128.55
kontena-agent | 
```

See https://github.com/weaveworks/weave/issues/2559

**Comment by SpComb** [comment]
Also happens on CoreOS alpha 1248.1.0

Leaves behind a dead docker container
```
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS                     PORTS               NAMES
72ca30c4f77f        weaveworks/weaveexec:1.7.2   "sh -c 'echo '# creat"   5 minutes ago       Dead                                           sharp_shirley
```

**Comment by jnummelin** [comment]
Fixed in #1569
